### PR TITLE
Add partialRefetch as an option

### LINKED ISSUE #128: partialRefetch as an option
The partialRefetch option is not available at the moment. Would it be possible to make it available?

More information here: https://www.apollographql.com/docs/react/essentials/queries

### PR DESCRIPTION
**Pull request by estrada9166**
fix #128 

Taken from [here](https://github.com/apollographql/react-apollo/blob/22f8ebf52b26b348d6be905d5b7fbbfea51c1541/src/Query.tsx#L455-L472)

### CODE REVIEW

**Comment by trojanowski** [comment]
Thanks @estrada9166. Could you add a unit test for it?

File: src/useQuery.ts [hunk 1]
```diff
@@ -163,6 +165,23 @@ export function useQuery<
...
 
+      // Taken from https://github.com/apollographql/react-apollo/blob/22f8ebf52b26b348d6be905d5b7fbbfea51c1541/src/Query.tsx#L455-L472
+      if (
+        partialRefetch &&
+        Object.keys(result.data).length === 0 &&
+        result.partial &&
+        fetchPolicy !== 'cache-only'
+      ) {
+        const partialRefetchResult = {
+          ...result.data,
```
Comment (brabeji) [inline src/useQuery.ts]: These are empty so maybe it would be better, to keep stale data from previous result using `useRef`.

Comment (brabeji) [reply src/useQuery.ts]: or maybe handle this here in a similar way as error is https://github.com/trojanowski/react-apollo-hooks/pull/159/files#diff-9affe4a75e3f039c2a00e27c8bf815aeR148

Comment (brabeji) [reply src/useQuery.ts]: BTW, sorry to invade this PR without introduction. I didn't catch it in time and coded basically the same thing yesterday evening. So now I'm just sharing my experience. Thanks for the PR!

Comment (estrada9166) [reply src/useQuery.ts]: Thanks for the comments! I'll update the PR with this! is much better.

Comment (brabeji) [reply src/useQuery.ts]: FWIW Also this one is related to the need of showing stale data:
https://github.com/trojanowski/react-apollo-hooks/pull/134

**Comment by trufa** [comment]
@brabeji Any new for this? :) Thanks!

**Comment by brabeji** [comment]
@trufa I'm using a locally patched version of `react-apollo-hooks` (via `patch-package`) and waiting for official support from `react-apollo`: https://github.com/apollographql/react-apollo/pull/2892#issuecomment-504544217

**Pull request closed without merging**